contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -60,41 +87,98 @@ procedure Lkt_Toolbox is
       return Decl_Node.P_As_Bare_Decl.Image;
    end Format_Node;
 
+   ----------------------------------
+   -- Print_Lkt_Toolbox_Diagnostic --
+   ----------------------------------
+
+   procedure Print_Lkt_Toolbox_Diagnostic
+     (Node : LK_Node'Class; Message : Wide_Wide_String)
+   is
+      Sloc_Range : constant Source_Location_Range := Node.Sloc_Range;
+      Unit       : constant Analysis.Analysis_Unit := Node.Unit;
+      Path       : constant String := Simple_Name (Unit.Get_Filename);
+   begin
+      Print_Diagnostic ((Sloc_Range, To_Unbounded_Text (Message)), Unit, Path);
+   end Print_Lkt_Toolbox_Diagnostic;
+
    ---------------------------
    -- Print_Semantic_Result --
    ---------------------------
 
    procedure Print_Semantic_Result
      (S : Analysis.Semantic_Result; Unit : Analysis.Analysis_Unit)
    is
+      Node : constant LK_Node'Class := Analysis.Node (S);
    begin
       if Analysis.Error_Message (S) /= "" then
          declare
             Diag : constant Diagnostic :=
-              (Analysis.Node (S).Sloc_Range,
+              (Node.Sloc_Range,
                To_Unbounded_Text (Analysis.Error_Message (S)));
          begin
+            --  Emit an error if the declaration including ``Node`` has no
+            --  ``@invalid`` annotation. Update ``Invalid_Decl_Map`` otherwise.
+
+            if Node.P_Topmost_Invalid_Decl.Is_Null then
+               Set_Exit_Status (1);
+
+               Print_Lkt_Toolbox_Diagnostic (Node,
+                  "unexpected diagnostic, is @invalid annotation missing?");
```

Comment:
Not quite: should be

```
Print_Lkt_Toolbox_Diagnostic
  (Node, "unexpected diagnostic, is @invalid annotation missing?");
```
In general: paren subexpressions should not go backwards in column numbers compared to the opening paren.

Reply:
Thanks! Hope it's good now!

Reply:
Yes, thanks!